core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+          CollectionLinearizerService.removeCollectionNode(
+              collection, 'exp_id0');
+          expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+          expect(
+              CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([]);
+        });
+
+    it('should correctly remove the first node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id0');
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, thirdCollectionNode]);
+    });
+
+    it('should correctly remove the last node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id2');
+      expect(collection.containsCollectionNode('exp_id2')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+    });
+
+    it('should correctly remove a middle node from a collection', function() {
+      var collection = _createLinearCollection();
+      CollectionLinearizerService.removeCollectionNode(collection, 'exp_id1');
+      expect(collection.containsCollectionNode('exp_id1')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode]);
+    });
+  });
+
+  describe('Collection linearizer node append function', function() {
+    it('should correctly append a node in an empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id0',
+          firstCollectionNode.getExplorationSummaryObject());
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([firstCollectionNode]);
+    });
+
+    it('should correctly append a node in a non-empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      collection.addCollectionNode(firstCollectionNode);
+      collection.addCollectionNode(secondCollectionNode);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id2',
+          thirdCollectionNode.getExplorationSummaryObject());
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
```

Comment:
Can you add a set of tests for ``getCollectionNodesInPlayableOrder``? I don't think you're testing it, but it's being used.